File: module/iOSLauncher/src/com/amz/ios/launcher/applibrary/AppLibraryAdapter.java
package com.amz.ios.launcher.applibrary;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.amz.ios.launcher.R;

import java.util.ArrayList;

public class AppLibraryAdapter extends RecyclerView.Adapter {

    ArrayList<AppCategory> mCategories;

    @NonNull
    @Override
    public RecyclerView.ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        LayoutInflater inflater = LayoutInflater.from(parent.getContext());
        View view = inflater.inflate(R.layout.apps_library_item,parent,false);
        return new ItemViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull RecyclerView.ViewHolder holder, int position) {

        if (holder instanceof ItemViewHolder) {
            ItemViewHolder itemViewHolder = (ItemViewHolder) holder;
            itemViewHolder.mItemFull.setApps(
                    mCategories.get(position).mApps
            );
            itemViewHolder.mItemFull.setTitle(
                    mCategories.get(position).mCategoryName
            );
        }

    }

    @Override
    public int getItemCount() {
        if (mCategories == null) return 0;
        return mCategories.size();
    }

    public class ItemViewHolder extends RecyclerView.ViewHolder {

        AppsLibraryItemFull mItemFull;

        public ItemViewHolder(@NonNull View itemView) {
            super(itemView);

            if (itemView instanceof AppsLibraryItemFull) {
                this.mItemFull = (AppsLibraryItemFull) itemView;
                mItemFull.setPadding(20,20,20,20);
            }
        }
    }
}

File: app/src/main/java/com/oslauncher/applauncher/themelauncher/Features/general/applibrary/adapters/AppLibraryAdapter.kt
package com.oslauncher.applauncher.themelauncher.Features.general.applibrary.adapters

import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView.Adapter
import androidx.recyclerview.widget.RecyclerView.ViewHolder
import com.oslauncher.applauncher.themelauncher.Features.general.applibrary.data.InstalledAppItem
import com.oslauncher.applauncher.themelauncher.databinding.ViewholderApplibAppBinding

/**
 * Danh sách app thật ở màn App Library: icon + tên + nhãn folder hiện tại + chevron.
 * Bấm 1 dòng -> [onSelectFolder] (mở Select Folder cho app đó).
 */
class AppLibraryAdapter(val listItem: ArrayList<InstalledAppItem>) : Adapter<ViewHolder>() {

    var onSelectFolder: ((InstalledAppItem) -> Unit)? = null

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        return AppVH(
            ViewholderApplibAppBinding.inflate(
                LayoutInflater.from(parent.context), parent, false
            )
        )
    }

    override fun getItemCount(): Int = listItem.size

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        val data = listItem[position]
        (holder as AppVH).binding.apply {
            ivAppIcon.setImageDrawable(data.icon)
            tvAppName.text = data.label
            tvFolderLabel.text = data.folderLabel
            root.setOnClickListener { onSelectFolder?.invoke(data) }
        }
    }

    class AppVH(val binding: ViewholderApplibAppBinding) : ViewHolder(binding.root)
}
